### DISCUSSION TITLE
Batch hybrid script

### DISCUSSION THREAD

**Discussion by God-damnit-all** [xonsh/xonsh#4931] [Q&A]
I'm interested in making a batch/xonsh hybrid script (aka polyglot script or chimera) but I'm having trouble coming up with a way to do it. To demonstrate what I'm after, I'll post a couple examples of how to do it for just batch/python

```
<# : ^
'''
@ECHO OFF
ECHO/This is batch code
python -x "%~f0" %*
EXIT /B %ERRORLEVEL%
'''
print('This is Python code')
```
OR
```
@SETLOCAL ENABLEDELAYEDEXPANSION & ECHO/This is batch code & python -x "%~f0" %* & EXIT /B !ERRORLEVEL!
print('This is Python code')
```
You can see an explanation of how these work in this Stack Overflow Q&A: https://stackoverflow.com/questions/17467441/how-to-embed-python-code-in-batch-script

Unfortunately, because xonsh doesn't have the -x parameter like python does, this doesn't really work, so I'm trying to determine if there's a good alternative.

**Comment by andry81** [comment]
> this doesn't really work

As I remember the Xonsh is rely on the `.xsh` file extension to run the xonsh parser first. And at last time I could not make it work even with the `.py` extension. So why just not put 2 files to the same directory and just run one from another?

**Reply by God-damnit-all** [reply]
> > this doesn't really work
> 
> As I remember the Xonsh is rely on the `.xsh` file extension to run the xonsh parser first. And at last time I could not make it work even with the `.py` extension. So why just not put 2 files to the same directory and just run one from another?

You must be mistaken, it did try to run the code when I had it call xonsh but, as expected, there were errors.

**Reply by andry81** [reply]
@ImportTaste https://github.com/xonsh/xonsh/issues/2478

**Reply by God-damnit-all** [reply]
> @ImportTaste #2478

Well, that's certainly one way of doing it. Kind of messy though.
